tests/standard/access_delegation.ts

```diff
@@ -1,59 +1,59 @@
 import helpers = require('../helpers');
 import * as AppsHelpers from "../apps_helpers";
 import webdriver = require('selenium-webdriver');
-import Bluebird = require('bluebird');
+import { decode } from "jsonwebtoken";
+import assert = require('assert');
 
 
 describe('access delegation', function () {
   it('should allow access to super admin', function () {
     this.timeout(10000);
 
-    return helpers.withChrome(function (driver) {
-      return Bluebird.resolve(driver.get(`http://john:password@localhost:3000/api/delegate`))
-        .then(function () {
-          return Bluebird.resolve(
-            driver.wait(webdriver.until.elementLocated(webdriver.By.css("p")), 5000))
-        })
-        .then(function (el) {
-          return Bluebird.resolve(driver.wait(webdriver.until.elementTextContains(el, 'This endpoint allows'), 5000));
-        });
+    return helpers.withChrome(async function (driver) {
+      await driver.get(`http://john:password@localhost:3000/api/delegate`);
+      const el = await driver.wait(webdriver.until.elementLocated(webdriver.By.css("p")), 5000);
+      return driver.wait(webdriver.until.elementTextContains(el, 'This endpoint allows'), 5000);
     });
   });
 
   it('should deny access to non super admin', function () {
     this.timeout(10000);
 
-    return helpers.withChrome(function (driver) {
-      return Bluebird.resolve(driver.get(`http://harry:password@localhost:3000/api/delegate`))
-        .then(function () {
-          return Bluebird.resolve(
-            driver.wait(webdriver.until.elementLocated(webdriver.By.css(".error-code")), 5000))
-        })
-        .then(function (el) {
-          return Bluebird.resolve(driver.wait(webdriver.until.elementTextContains(el, 'HTTP ERROR 403'), 5000));
-        });
+    return helpers.withChrome(async function (driver) {
+      await driver.get(`http://harry:password@localhost:3000/api/delegate`);
+      const el = await driver.wait(webdriver.until.elementLocated(webdriver.By.css(".error-code")), 5000);
+      return driver.wait(webdriver.until.elementTextContains(el, 'HTTP ERROR 403'), 5000);
     });
   });
 
+  it('should generate token with correct expiration', async function () {
+    this.timeout(10000);
+    const instanceId = this.mesosTaskIds['app1'];
+
+    let token = await AppsHelpers.generateAccessToken('john', 'harry', instanceId, '1h');
+    let json = decode(token) as any;
+    assert.equal(json.exp, json.iat + 3600);
+
+    token = await AppsHelpers.generateAccessToken('john', 'harry', instanceId, '1d');
```

Comment:
Fixture is different, you should create another test for this, or make this test generic so that it could be called for multiple (input, expected output) values

Reply:
done